### PR TITLE
1.79

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Tooling

- Updates Rust to 1.79

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [x] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

**Review by fbjork — APPROVED** [review]

File: cli/crates/backend/src/api/graphql/types/mutations/environment_variable_delete.rs [hunk 1]
```diff
@@ -47,7 +47,9 @@ pub struct EnvironmentVariableDeleteByValuesSuccess {
 
 #[derive(cynic::InlineFragments, Debug)]
 pub enum EnvironmentVariableDeleteByValuesPayload {
+    #[allow(dead_code)]
     EnvironmentVariableDeleteByValuesSuccess(EnvironmentVariableDeleteByValuesSuccess),
+    #[allow(dead_code)]
```
Comment (obmarg) [inline cli/crates/backend/src/api/graphql/types/mutations/environment_variable_delete.rs:52]: These seem like they _should_ be used, no?  Or if we don't need them, should they be removed?

File: engine/crates/parser-sdl/src/rules/visitor/context.rs [hunk 2]
```diff
@@ -45,18 +45,21 @@ pub struct VisitorContext<'a> {
...
     /// schema (such as global headers), but *MUST NOT* be used for anything that requires a stable
-    /// identifier across schema generations.
+    /// identifier across schema generations.    
+    #[allow(dead_code)]
     pub connector_id_generator: ConnectorIdGenerator,
 
     /// Each schema to build should contains a SchemaID -> MetaType String to be
     /// able to construct the whole SchemaRegistry at the end of the parsing.
+    #[allow(dead_code)]
     pub schema_to_build: RwLock<HashMap<SchemaID, String>>,
```
Comment (obmarg) [inline engine/crates/parser-sdl/src/rules/visitor/context.rs:63]: If these aren't used maybe we should delete them

**Review by obmarg — COMMENTED** [review]
Should we should be taking advantage of the improved dead_code detection to remove/fix unused things instead of just marking them all as allowed?

I think some of them make sense to mark as allow, but others look quite suspect to me

**Comment by yoav-lavi** [comment]
@obmarg normally I'd go over these individually and remove them, but I won't have the time to today with the migration and I don't want to stall the upgrade. We should likely remove them in the future or check why they're not used

File: cli/crates/backend/src/api/graphql/types/mutations/environment_variable_delete.rs (hunk 1 above)
Comment (fbjork) [reply cli/crates/backend/src/api/graphql/types/mutations/environment_variable_delete.rs:52]: Yeah let's delete all dead code

File: cli/crates/backend/src/api/graphql/types/mutations/environment_variable_upsert_by_slugs.rs [hunk 3]
```diff
@@ -28,13 +28,15 @@ pub struct NameContainsInvalidCharactersError {
 #[derive(cynic::QueryFragment, Debug)]
 #[cynic(graphql_type = "Mutation", variables = "EnvironmentVariableUpsertBySlugsVariables")]
 pub struct EnvironmentVariableUpsertBySlugs {
+    #[allow(dead_code)]
```
Comment (fbjork) [inline cli/crates/backend/src/api/graphql/types/mutations/environment_variable_upsert_by_slugs.rs:31]: Can't we delete the dead code?

Comment (pimeys) [reply cli/crates/backend/src/api/graphql/types/mutations/environment_variable_upsert_by_slugs.rs:31]: Another PR. There's lots of it, so just removed the things Yoav added in this PR.

**Pull request merged**